File: src/Easy/FindFixedPoint/Python/find_fixed_point.py
import sys
sys.path.insert(1, sys.path[0] + '/../../../RunTests/Python')
from run_test import run_tests

def find_fixed_point_for_loop(numbers):
  """
  Find a fixed point in an array using a traditional for loop and the enumerate function

  Arguments:
      numbers { number[] } -- An array of sorted integers

  Returns:
      number -- The index of the first fixed point, or -1 if there is none
  """
  for index, value in enumerate(numbers):
    if index == value:
      return value
  return -1

run_tests( find_fixed_point_for_loop )

File: src/Easy/FindFixedPoint/JavaScript/FindFixedPoint.js
const runTests = require('../../../RunTests/JavaScript/index')

/**
 * Solution using Array.prototype.some()
 * A function that returns a fixed point in a sorted list of integer
 * @param { number[] } numbers -  An array of sorted integers 
 * @return { number | null} match - The index of the fixed point, or null if there is none 
 */
function findFixedPointArraySome( numbers ){
	let match
  numbers.some( ( number, index ) => match = number === index ? number + 1 : match)
	return match ? match - 1 : -1
}

/**
 * Solution using a traditional for loop
 * A function that returns a fixed point in a sorted list of integer
 * @param { number[] } numbers -  An array of sorted integers 
 * @return { number | null} match - The index of the fixed point, or null if there is none 
 */
function findFixedPointForLoop( numbers ) {
  for ( let i = 0; i < numbers.length; i++) {
    if ( i === numbers[i] ) return i
  }
  return -1
}

runTests( findFixedPointArraySome )
runTests( findFixedPointForLoop )

